kmir/src/kmir/kdist/mir-semantics/kmir.md

```diff
@@ -581,13 +581,17 @@ Otherwise the provided message is passed to a `panic!` call, ending the program
            #expect(COND, EXPECTED, MSG) ~> #execBlockIdx(TARGET)
        </k>
 
-  syntax KItem ::= #expect ( Evaluation, Bool, AssertMessage ) [strict(1)]
+  syntax Evaluation ::= #expect ( Evaluation, Bool, AssertMessage ) [strict(1)]
 
   rule <k> #expect(typedValue(BoolVal(COND), _, _), EXPECTED, _MSG) => .K ... </k>
     requires COND ==Bool EXPECTED
 
   rule <k> #expect(typedValue(BoolVal(COND), _, _), EXPECTED, MSG) => AssertError(MSG) ... </k>
     requires COND =/=Bool EXPECTED
+
+  rule <k> thunk(#expect(_, _, _MSG)) => .K ... </k>
+
+  rule <k> thunk(#expect(_, _, MSG)) => AssertError(MSG) ... </k>
```

Comment:
Add some documentation here, before this code-block, indicating why we add these rules. Specifically:

- if one of the more specific rules for `#expect` fires, we know that we _definitely don't_ have an assertion failure, or that we _definitely do_ have an assertion failure.
- Otherwise, the `thunk` wrapper indicates that we don't know, and so instead we non-deterministically explore both branches. This will be sound for our proofs (we won't lose assertion failures, just may have too many).